internal/cache/file/cache_handler.go

```diff
@@ -0,0 +1,181 @@
+	key := fileInfo.Key
+	_, err := key.Key()
+	if err != nil {
+		return fmt.Errorf("error while performing post eviction: %v", err)
+	}
+
+	chr.jobManager.RemoveJob(key.ObjectName, key.BucketName)
+
+	localFilePath := util.GetDownloadPath(chr.cacheLocation, util.GetObjectPath(key.BucketName, key.ObjectName))
+	err = os.Remove(localFilePath)
+	if err != nil {
+		return fmt.Errorf("while deleting file: %s, error: %v", localFilePath, err)
+	}
+
+	return nil
+}
+
+// Acquires and releases LOCK(CacheHandler.mu)
+func (chr *CacheHandler) addFileInfoEntryInTheCacheIfNotAlready(object *gcs.MinObject, bucket gcs.Bucket) error {
+	fileInfoKey := data.FileInfoKey{
+		BucketName: bucket.Name(),
+		ObjectName: object.Name,
+	}
+	fileInfoKeyName, err := fileInfoKey.Key()
+	if err != nil {
+		return fmt.Errorf("while creating key: %v", fileInfoKeyName)
+	}
+
+	chr.mu.Lock()
+	defer chr.mu.Unlock()
+
+	// Todo - (raj-prince) - this lookUp should not change the LRU order.
+	fileInfo := chr.fileInfoCache.LookUp(fileInfoKeyName)
+	if fileInfo == nil {
+		fileInfo = data.FileInfo{
+			Key:              fileInfoKey,
+			ObjectGeneration: object.Generation,
+			Offset:           0,
+			FileSize:         object.Size,
+		}
+
+		evictedValues, err := chr.fileInfoCache.Insert(fileInfoKeyName, fileInfo)
+		if err != nil {
+			return fmt.Errorf("while inserting into the cache: %v", err)
+		}
+
+		for _, val := range evictedValues {
+			fileInfo := val.(data.FileInfo)
+			err := chr.performPostEvictionWork(&fileInfo)
+			if err != nil {
+				return fmt.Errorf("while performing post eviction of %s object error: %v", fileInfo.Key.ObjectName, err)
+			}
+		}
+	} else {
+		job := chr.jobManager.GetJob(object, bucket)
+		jobStatus := job.GetStatus()
+		if jobStatus.Name != downloader.FAILED {
+			// Move this entry on top of LRU.
+			_ = chr.fileInfoCache.LookUp(fileInfoKeyName)
+		}
+	}
+
+	return nil
+}
+
+// GetCacheHandle creates an entry in fileInfoCache if it does not already exist.
+// It creates FileDownloadJob if not already exist. Also, creates localFilePath
+// which contains the downloaded content. Finally, it returns a CacheHandle that
+// contains the async DownloadJob and the local file handle.
+// Acquires and releases LOCK(CacheHandler.mu)
+func (chr *CacheHandler) GetCacheHandle(object *gcs.MinObject, bucket gcs.Bucket, initialOffset int64) (*CacheHandle, error) {
+	err := chr.addFileInfoEntryInTheCacheIfNotAlready(object, bucket)
+	if err != nil {
+		return nil, fmt.Errorf("while adding the entry in the cache: %v", err)
+	}
+
+	localFileReadHandle, err := chr.createLocalFileReadHandle(object.Name, bucket.Name())
+	if err != nil {
+		return nil, fmt.Errorf("while create local-file read handle: %v", err)
+	}
+
+	return NewCacheHandle(localFileReadHandle, chr.jobManager.GetJob(object, bucket), chr.fileInfoCache, initialOffset), nil
+}
+
+// InvalidateCache removes the entry from the fileInfoCache, cancel the async running job incase,
```

Comment:
nit: removes download job and deletes the local file in cache for given object.
nit: Remove the TODO.